test/unit/nic_test.rb

```diff
@@ -113,6 +120,99 @@ def teardown
         @interface.proxy
       end
     end
+
+    test "we can't destroy primary interface of managed host" do
+      host = FactoryGirl.create(:host, :managed, :ip => '127.0.0.1')
```

Comment:
This host creation (repeated in all tests) should be moved to a setup block in two contexts, one for managed hosts and another for unmanaged hosts.